File: Código/Loja/src/banco/OperacoesBanco.java
package banco;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

import dados.Cliente;
import dados.Produto;

public class OperacoesBanco {

    private int quantClientes, quantProdutos, quantFornecedores, quantFucionarios;
    private Connection conexao = null;
    private PreparedStatement stmt = null;
    private ResultSet rs = null;
    private String sql;

    public boolean inserirCliente(Cliente cliente) {
        sql = "INSERT INTO Cliente (nome, cpf, rg, idade) VALUES (?, ?, ?, ?)";
        try {
            conexao = ConexaoBanco.getConnection();
            stmt = conexao.prepareStatement(sql);
            stmt.setString(1, cliente.getNome());
            stmt.setString(2, cliente.getCpf());
            stmt.setString(3, cliente.getRg());
            int linhasAfetadas = stmt.executeUpdate();
            quantClientes++;
            return linhasAfetadas > 0;
        } catch (SQLException e) {
            if (e.getMessage().contains("Duplicate") || e.getErrorCode() == 1062) {
                System.out.println("Cliente já existe no banco de dados: " + e.getMessage());
            } else {
                System.out.println("Erro ao inserir cliente: " + e.getMessage());
            }
            e.printStackTrace();
            return false;
        } finally {
            try {
                if (stmt != null)
                    stmt.close();
                if (conexao != null)
                    conexao.close();
            } catch (SQLException e) {
                System.out.println("Erro ao fechar recursos: " + e.getMessage());
            }
        }
    }

    public boolean inserirProduto(Produto produto) {
    }

    public List<Cliente> consultarClientesPorNome(String nome) {
        List<Cliente> clientes = new ArrayList<>();
        sql = "SELECT * FROM Cliente WHERE nome LIKE ?";
        try {
            conexao = ConexaoBanco.getConnection();
            stmt = conexao.prepareStatement(sql);
            stmt.setString(1, "%" + nome + "%");
            rs = stmt.executeQuery();
            while (rs.next()) {
                Cliente cliente = new Cliente();
                cliente.setId(rs.getInt("id_cliente"));
                cliente.setNome(rs.getString("nome"));
                cliente.setCpf(rs.getString("cpf"));
                cliente.setRg(rs.getString("rg"));
                cliente.setIdade(rs.getInt("idade"));
                clientes.add(cliente);
            }
        } catch (SQLException e) {
            System.out.println("Erro ao consultar clientes: " + e.getMessage());
            e.printStackTrace();
        } finally {
            try {
                if (rs != null)
                    rs.close();
                if (stmt != null)
                    stmt.close();
                if (conexao != null)
                    conexao.close();
            } catch (SQLException e) {
                System.out.println("Erro ao fechar recursos: " + e.getMessage());
            }
        }
        return clientes;
    }

    public int getQuantidadeClientes() {
        return quantClientes;
    }

}

File: Código/Python/Loja/src/banco/OperacoesBanco.py
import mysql.connector
from mysql.connector import Error
from banco.conexao_banco import ConexaoBanco
from dados.cliente import Cliente

class OperacoesBanco:
    def __init__(self):
        self.quant_clientes = 0
        self.quant_produtos = 0
        self.quant_fornecedores = 0
        self.quant_funcionarios = 0

    def inserir_cliente(self, cliente: Cliente):
        sql = "INSERT INTO Cliente (nome, cpf, rg, idade) VALUES (%s, %s, %s, %s)"
        try:
            conexao = ConexaoBanco.get_connection()
            if conexao is None:
                print("Erro ao conectar ao banco de dados")
                return False
            
            cursor = conexao.cursor()
            cursor.execute(sql, (cliente.nome, cliente.cpf, cliente.rg, cliente.idade))
            conexao.commit()
            self.quant_clientes += 1
            return cursor.rowcount > 0

        except Error as e:
            if "Duplicate" in str(e) or e.errno == 1062:
                print(f"Cliente já existe no banco de dados: {e}")
            else:
                print(f"Erro ao inserir cliente: {e}")
        finally:
            if cursor:
                cursor.close()
            if conexao:
                conexao.close()

        return False

    def consultar_clientes_por_nome(self, nome):
        sql = "SELECT * FROM Cliente WHERE nome LIKE %s"
        clientes = []
        try:
            conexao = ConexaoBanco.get_connection()
            if conexao is None:
                print("Erro ao conectar ao banco de dados")
                return []

            cursor = conexao.cursor(dictionary=True)
            cursor.execute(sql, (f"%{nome}%",))
            for row in cursor.fetchall():
                cliente = Cliente(row["nome"], row["cpf"], row["rg"])
                cliente.id = row["id_cliente"]
                cliente.idade = row["idade"]
                clientes.append(cliente)

        except Error as e:
            print(f"Erro ao consultar clientes: {e}")
        finally:
            if cursor:
                cursor.close()
            if conexao:
                conexao.close()

        return clientes

    def get_quantidade_clientes(self):
        return self.quant_clientes